### ISSUE TITLE
Nanominer should restart when shares continue to be rejected

### ISSUE THREAD

**Issue by wtu-fl** [nanopool/nanominer#70]
Often after a new release of Nanominer is made available, some parameters must be changed between the client and the Nanopool server because I frequently get this:
2020-02-17 10:37:34: <info>  Ethereum - SHARE FOUND (GPU: 2, nonce: 0x4eeb7fc8f8ab6de6).
2020-02-17 10:37:34: <info> Ethereum: share denied: Wrong params (157 ms).
2020-02-17 10:37:43: [Statistics] Ethereum - Total speed: 159.212 MH/s, Total shares: 5021 Rejected: 15, Time: 3d 16h 41m 30s
This can go on for hours and hours, the rig using the usual amount of power, until I happen to wander over and discover it.
Restarting Nanominer with autoUpdate=true always solves the problem.  If only Nanominer could automatically detect several Rejected shares in a row as a problem, and restart itself, that one improvement would probably improve my net returns by 10 or 20%... because this seems to happen an average of once or twice a week.
Thanks!

**Comment by Grumpy-Dwarf** [comment]
There used to be a bug #67 with protocol detection. Please check if this still happens with latest nanominer 1.8.2.

**Comment by wtu-fl** [comment]
It hasn't happened (yet) since the upgrade to 1.8.2, but it has only been about 24 hours.  I will advise if it happens again.

Usually it would happen right after a new Nanominer version was released, but before I knew to install it… in other words, when the prior version had continued running for several days, then suddenly this would happen.  That is when I discovered a new version had just been released, and re-starting Nanominer would solve it.

This made me think something had been changed on the Nanopool server side concurrent with the new release of the Nanominer client.
